pubtools/_pulp/tasks/set_maintenance/base.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from pubtools._pulp.task import PulpTask
+from pubtools._pulp.services import PulpClientService
+
+
+step = PulpTask.step
+LOG = logging.getLogger("set-maintenance")
+
+
+class SetMaintenance(PulpClientService, PulpTask):
+    def add_args(self):
+        super(SetMaintenance, self).add_args()
+
+        self.parser.add_argument("--owner", help="who sets/unsets maintenance mode")
+
+        self.parser.add_argument(
```

Comment:
You might want to consider adding a `type` here which compiles the regex, the main benefit being that the code will fail as early as possible if the user passes something which is not a valid regex.  Without this, I guess it won't fail until part way through the task.  A minor suggestion only.